custom_components/kia_uvo/__init__.py

```diff
@@ -244,12 +262,29 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry):
         )
     )
     if unload_ok:
-        vehicle_topic_listener = hass.data[DOMAIN][DATA_VEHICLE_LISTENER]
-        vehicle_topic_listener()
+        vehicle_identifier = config_entry.data[CONF_VEHICLE_IDENTIFIER]
+        vehicle_listener = hass.data[DOMAIN][vehicle_identifier][DATA_VEHICLE_LISTENER]
+        vehicle_listener()
 
-        config_update_listener = hass.data[DOMAIN][DATA_CONFIG_UPDATE_LISTENER]
+        config_update_listener = hass.data[DOMAIN][vehicle_identifier][
+            DATA_CONFIG_UPDATE_LISTENER
+        ]
         config_update_listener()
 
-        hass.data[DOMAIN] = None
+        hass.data[DOMAIN][vehicle_identifier] = None
 
     return unload_ok
+
+
+async def async_migrate_entry(hass, config_entry: ConfigEntry):
+    if config_entry.version == 1:
+
+        vehicle_id = config_entry.data["stored_credentials"]["vehicle_id"]
+        new_data = config_entry.data.copy()
+        new_data[CONF_VEHICLE_IDENTIFIER] = vehicle_id
```

Comment:
looking here, vehicle_id should be equal to device_id. So, are these equal?
`device_registry.async_get(device_id)` to `vehicle_id`. I personally do not know.